internal/profiling/buffer_test.go

```diff
@@ -0,0 +1,188 @@
+// +build !appengine
+
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package profiling
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestCircularBufferSerial(t *testing.T) {
+	var size, i uint32
+	var result []interface{}
+
+	size = 1 << 15
+	cb, err := newCircularBuffer(size)
+	if err != nil {
+		t.Errorf("error allocating CircularBuffer: %v", err)
+		return
+	}
+
+	for i = 0; i < size/2; i++ {
+		cb.Push(i)
+	}
+
+	result = cb.Drain()
+	if uint32(len(result)) != size/2 {
+		t.Errorf("expected result size %d, got %d", size/2, len(result))
```

Comment:
Error message should usually have the following format:
```foobar(args) = <whatever you got>, want <whatever you want>```

For example, this is an excerpt from the godoc for the testing package:
```golang
func TestAbs(t *testing.T) {
    got := Abs(-1)
    if got != 1 {
        t.Errorf("Abs(-1) = %d; want 1", got)
    }
}
```

Reply:
Done.